### DISCUSSION TITLE
How to create arrow auto position?

### DISCUSSION THREAD

**Discussion by yurenklimbu** [floating-ui/floating-ui#2819] [Q&A]
Hi, I'm trying to create a popover with an auto-positioning arrow element, but I'm unable to fix it.

https://stackblitz.com/edit/vitejs-vite-vfs8n5?file=src%2FApp.vue,src%2Fcomponents%2Fdropdown.vue&terminal=dev

**Comment by atomiks** [comment]
I'm not sure what the problem is as the arrow is being correctly positioned on the x-axis in the demos. Maybe you want the `shift()` middleware to prevent the popover from leaving the edges of the screen, whereby the arrow will still centered to the reference in your demo. Or, you want the y-axis to be automatically positioned, which the data doesn't provide since it's not necessary - you can do it with plain CSS using the returned `placement`.

**Reply by yurenklimbu** [reply]
@atomiks Thanks!

I'm trying to make if floating target element placement bottom arrow should be in bottom position but it arrow always positioned in top center same as left or right it always positioned left center.

**Reply by oom-** [reply]
I'm thinking that @yurenklimbu speaks about the y-axis placement.
I do not understand your answer @atomiks : 
> which the data doesn't provide since it's not necessary

I mean, in the documentation it's writing to be sure to add the arrow middleware at the end to get the best position calculation.
We have to provide the "style" with the x and y position of the arrow, so why only one axis at a time is used ?

It would make much more sense that the arrow get the correct x-axis / y-axis position from out of the box.


Edit: @yurenklimbu 
I solved it this way in vue: 
```javascript
const styleArrow = computed(() => {
    const isLeft = /left/.test(placement.value);
    const isRight = /right/.test(placement.value);
    const isTop = /top/.test(placement.value);
    const isBottom = /bottom/.test(placement.value);
    return {
        left: isBottom || isTop ? `${middlewareData.value.arrow?.x}px` : isRight ? 0 : "",
        right: isBottom || isTop ? `${middlewareData.value.arrow?.x}px` : isLeft ? 0 : "",
        top: isLeft || isRight ? `${middlewareData.value.arrow?.y}px` : isBottom ? 0 : "",
        bottom: isLeft || isRight ? `${middlewareData.value.arrow?.y}px` : isTop ? 0 : ""
    };
});
```
